AutoCollection/WebSnippet.ts

```diff
@@ -0,0 +1,146 @@
+            if (originalRequestListener) {
+                requestListener = (request: http.IncomingMessage, response: http.ServerResponse) => {
+                    // Patch response write method
+                    let originalResponseWrite = response.write;
+                    let isEnabled = this._isEnabled;
+                    response.write = function wrap(a: Buffer | string, b?: Function | string, c?: Function | string) {
+                        if (isEnabled) {
+                            if (WebSnippet.ValidateInjection(response, a)) {
+                                arguments[0] = WebSnippet.InjectWebSnippet(response, a);
+                            }
+                        }
+                        return originalResponseWrite.apply(response, arguments);
+                    }
+                    // Patch response end method for cases when HTML is added there
+                    let originalResponseEnd = response.end;
+
+                    response.end = function wrap(a?: Buffer | string | any, b?: Function | string, c?: Function) {
+                        if (isEnabled) {
+                            if (WebSnippet.ValidateInjection(response, a)) {
+                                arguments[0] = WebSnippet.InjectWebSnippet(response, a);
+                            }
+                        }
+                        originalResponseEnd.apply(response, arguments);
+                    }
+
+                    originalRequestListener(request, response);
+                }
+            }
+            return originalHttpServer(requestListener);
+        }
+    }
+
+    /**
+     * Validate response and try to inject Web snippet
+     */
+    public static ValidateInjection(response: http.ServerResponse, input: string | Buffer): boolean {
+        if (response && input) {
+            // insert snippet if only response returns 200
+            if (response.statusCode != 200) return false;
+            let contentType = response.getHeader('Content-Type');
+            let contentEncoding = response.getHeader('Content-Encoding'); // Compressed content not supported for injection
+            if (!contentEncoding && contentType && typeof contentType == "string" && contentType.toLowerCase().indexOf("text/html") >= 0) {
+                let html = input.toString();
+                if (html.indexOf("<head>") >= 0 && html.indexOf("</head>") >= 0) {
+                    // Check if snippet not already present looking for AI Web SDK URL
+                    if (html.indexOf(WebSnippet._aiUrl) < 0) {
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Inject Web snippet
+     */
+    public static InjectWebSnippet(response: http.ServerResponse, input: string | Buffer): string | Buffer {
+        try {
+            let hasContentHeader = !!response.getHeader('Content-Length');
```

Comment:
hasContentLengthHeader would be a good name

Reply:
updated